src/core/Track.cpp

```diff
@@ -386,7 +466,7 @@ QColor TrackContentObjectView::mutedBackgroundColor() const
 { return m_mutedBackgroundColor; }
 
 QColor TrackContentObjectView::selectedColor() const
-{ return m_selectedColor; }
+{ return ( m_usesCustomSelectedColor ? m_customSelectedColor : m_selectedColor ); }
```

Comment:
I agree that it is a function to retrieve the `selectedColor` property, however I found it to be more convenient to check whether there is a custom selection color (i.e. the TCO has a custom color) here, in this function, rather than have to implement logic for checking whether the track has a color every time `selectedColor()` is called (implement logic once vs. atleast 4 times). Using another function to get the custom selection color leaves space for some possible bugs, and still has the problem that there now has to be logic every time `selectedColor()` is called.

I'm against changing this back to original state, and vouch for allowing an exception for this, but if you think that there's no point or there's a better way, please let me know.

Reply:
My concern is that this leads to unintuitive behaviour for this function, and possibly bugs as a consequence. Consider a few cases:
```cpp
TrackContentObjectView *tcoView = getMeATcoView();
tcoView->setSelectedColor(QColor{11, 22, 33});
tcoView->selectedColor(); // <- != QColor(11, 22, 33) - where did my colour go!?
```
```cpp
TrackContentObjectView *tcoView = getMeATcoView();
tcoView->setSelectedColor(tcoView->selectedColor().darker(120)); // Dim the selected colour a bit
// Later on, user removes custom colour from clip
// Clip now has a broken selected colour, based on an old custom colour
```
What used to be a public accessor function - and is still explicitly declared to be a public accessor function in a `Q_PROPERTY` in `Track.h` - has been repurposed as an internal utility function. This can lead to bugs when this function is used by someone who thinks it will behave sanely, and isn't aware of its actual purpose. Furthermore, given that this is a property getter, one of those users is Qt, and we don't have control over that code. I would rather you kept the property getter as a property getter, and if you need a utility function to get a colour to paint with, add a new one.

Reply:
Alright, so I rewrote logic in all the `paintEvent`s to make them darken the TCO color appropriately while selected. Due to that, I've got some pros and cons.

Pros:
- Your concern is solved, and the `selectedColor()` function does as expected.
- I've noticed that there's no point in storing a custom selection color anyways, as it is always equal to `color.darker(200)`.

Cons:
- There's a lot more logic involved in the `paintEvent`s, turning what's essentially 4-5 lines of code into ~20 just in the name of readability.
- There's a lot of magic numbers (i.e. constants) which aren't synced across `paintEvent`s, thus if a dev is unaware of this, selected colors and muted colors will be inconsistent across different types of TCOs.

And as a side note: I also changed code such that TCOs when muted also get colored in a darker variant of their assigned custom color (non-colored TCOs work as always).

![image](https://user-images.githubusercontent.com/26451259/93014532-70ef7c80-f5cf-11ea-9724-efee396d2aa5.png)
_Muted tracks_

![image](https://user-images.githubusercontent.com/26451259/93014538-82388900-f5cf-11ea-9d48-490100678e89.png)
_Unmuted tracks_

I think I should post this as a comment of its own xD

Reply:
In response to the downsides, I would quote Dom from the comment immediately above yours:

> if you need a utility function to get a colour to paint with, add a new one.

This should solve the issues mentioned.

Reply:
Yeah, and I think since I changed the muted color logic too, I should add a function for that as well in the superclass. I'll be doing that soon.

Reply:
I've added a `getColorForDisplay()` function by the latest commit, if that is satisfactory, you may resolve this comment.